projects/igniteui-angular/src/lib/toast/toast.component.ts

```diff
@@ -331,49 +371,28 @@ export class IgxToastComponent implements IToggleView, OnInit, OnDestroy {
      * @memberof IgxToastComponent
      */
     public toggle() {
-        this.isVisible ? this.close() : this.open();
-    }
-
-    /**
-     * @hidden
-     */
-    @HostBinding('class.igx-toast--top')
-    public get cssClassTop() {
-       return this.position === IgxToastPosition.Top;
-    }
-
-    /**
-     * @hidden
-     */
-    @HostBinding('class.igx-toast--middle')
-    public get cssClassMiddle() {
-       return this.position === IgxToastPosition.Middle;
-    }
-
-    /**
-     * @hidden
-     */
-    @HostBinding('class.igx-toast--bottom')
-    public get cssClassBottom() {
-       return this.position === IgxToastPosition.Bottom;
+        super.toggle();
     }
 
     /**
      * @hidden
      */
-    public ngOnInit() {
-        if (this.navService && this.id) {
-            this.navService.add(this.id, this);
-        }
+    ngOnInit() {
+        this.onAppended.pipe(takeUntil(this.d$)).subscribe(() => {
```

Comment:
`onAppended` is emitted when the content is moved to the overlay's content `div`. At this point content is still not visible, neither it is shown. Why not subscribe to `onOpened`?